lib/utils/documentdb.ts

```diff
@@ -263,6 +263,65 @@ export function readDocument<T>(
   });
 }
 
+/**
+ * Wraps executeNext inner QueryIterator method in a Promise
+ *
+ * Returns an Either:
+ *
+ *    right(some(RetrievedDocuments)): when documents exists and are correctly retrieved
+ *    right(none): in case the iterator haven't found any document
+ *    left(QueryError): in case of error querying the database
+ */
+function executeNext<T>(
```

Comment:
this is the same method as before, but since it's now used by two different methods I've extracted it to its own function